examples/dat3m/kit/bench.py

```diff
@@ -0,0 +1,235 @@
+
+        self.dat3m_memorymodels = [
+            mm[:-4]
+            for elem in self.platform.comm.shell(command="ls -1 cat").splitlines()
+            if (mm := elem.strip()).endswith(".cat")
+        ]
+        self.dat3m_locks = [
+            lock[:-2]
+            for elem in self.platform.comm.shell(command="ls -1 benchmarks/locks").splitlines()
+            if (lock := elem.strip()).endswith(".c")
+        ]
+        self.dat3m_targets = [
+            "c11",
+            "arm8",
+            "power",
+            "ptx",
+            "tso",
+            "imm",
+            "lkmm",
+            "riscv",
+            "vulkan",
+        ]
+        self._tmp_results = {}
+
+    @property
+    def bench_src_path(self) -> pathlib.Path:
+        return self._bench_src_path
+
+    @staticmethod
+    def get_build_var_names() -> List[str]:
+        return []
+
+    @staticmethod
+    def get_run_var_names() -> List[str]:
+        return [
+            "memory_model",
+            "target_arch",
+            "lock_name",
+            "bound",
+        ]
+
+    @staticmethod
+    def get_tilt_var_names() -> List[str]:
+        return []
+
+    def dependencies(self) -> List[PackageDependency]:
+        return super().dependencies() + []
+
+    def build_tilt(self, **kwargs) -> None:
+        self.tilt.build_single_lock(**kwargs)
+
+    def prebuild_bench(
+        self,
+        **_kwargs,
+    ) -> None:
+        pass
+
+    def build_bench(
+        self,
+        **kwargs,
+    ) -> None:
+        pass
+
+    def clean_bench(self) -> None:
+        pass
+
+    def single_run(  # pylint: disable=arguments-differ
+        self,
+        benchmark_duration_seconds: int,
+        memory_model: str = "aarch64",
+        target_arch: str = "arm8",
+        lock_name: str = "ttas",
+        bound: int | None = None,
+        **kwargs,
+    ) -> str:
+        # Sanitize input variables
+        if memory_model not in self.dat3m_memorymodels:
+            raise ValueError(
+                f"Unknown memory model: {memory_model}\nSupported memory models: {self.dat3m_memorymodels}"
+            )
+        if target_arch not in self.dat3m_targets:
+            raise ValueError(
+                f"Unknown target architecture: {target_arch}\nSupported targets: {self.dat3m_targets}"
+            )
+        if lock_name not in self.dat3m_locks:
+            raise ValueError(f"Unknown lock name: {lock_name}\nSupported locks: {self.dat3m_locks}")
+
+        # Generate environment
+        environment = self._preload_env(
+            benchmark_duration_seconds=benchmark_duration_seconds,
+            memory_model=memory_model,
+            target_arch=target_arch,
+            lock_name=lock_name,
+            bound=bound,
+            **kwargs,
+        )
+
+        # Generate dat3m command line using variables
+        command_head = [
+            "java",
+            "-jar",
+            "dartagnan/target/dartagnan.jar",
```

Comment:
This is not just intended to be run using the docker containers, right? The user could run benchkit from host if dat3m is installed. If so, maybe you should prefix every file/folder related to dat3m with `DAT3M_HOME`

Reply:
indeed, but i need to support retrieving value in environment variable through the comm layer (either docker or the host). That can be done.